### DISCUSSION TITLE
Visualize a Schedule

### DISCUSSION THREAD

**Discussion by drusellers** [quartznet/quartznet#2208] [Q&A] [answered]
Is there a way I can take a trigger and have it generate the times it will run? Basically, I want to "check my math".

Specifically, I'm working on getting a job to run once per day at 3am Eastern.

```csharp
        var autoSubmitTrigger = TriggerBuilder.Create()
            .WithPriority(10)
            .ForJob(AutoSubmitOrdersJob.Key)
            // 3am ET ~= 7am UTC
            .WithDailyTimeIntervalSchedule(x =>
                x.OnEveryDay().StartingDailyAt(TimeOfDay.HourAndMinuteOfDay(7,0))
                    .WithRepeatCount(1)
                )
            .StartNow()
            .WithIdentity("auto-submit-daily", "orders")
            .Build();
```

It seems that the way I have it setup it repeats forever once it starts. 🤔

**Comment by lahma** [comment]
You should be able use `TriggerUtils` and its helpers for this like `TriggerUtils.ComputeFireTimes()` and `TriggerUtils.ComputeFireTimesBetween()`.

**Reply by drusellers** [reply]
```csharp
        var autoSubmitTrigger = TriggerBuilder.Create()
            .WithPriority(10)
            .ForJob(AutoSubmitOrdersJob.Key)
            // 3am ET ~= 7am UTC
            .WithDailyTimeIntervalSchedule(x =>
                x.OnEveryDay().StartingDailyAt(TimeOfDay.HourAndMinuteOfDay(7,0))
                    .WithRepeatCount(1)
            )
            .StartNow()
            .WithIdentity("auto-submit-daily", "orders")
            .Build() as DailyTimeIntervalTriggerImpl;

        var x = TriggerUtils.ComputeFireTimes(autoSubmitTrigger, new WeeklyCalendar(), 8);
```

But then I only get two items back, which are like 6:03pm and 6:04pm (the 6pm part is great) the 03 and 04 are whatever. What is interesting is that I only get two items back. I'm expecting to see 1 fire time for each day, and then get back 8 fire times.

But I'm guessing I have a misunderstanding in how this all works.

**Reply by lahma** [reply, marked as answer]
Well daily interval scheduling is meant more like for "with interval of X during this daily window". So you could say that trigger runs "MON,WED,FRI every five minutes between 8am-4pm".

Your description sounds more like a simple cron expression:

```c#
var trigger = (IOperableTrigger) TriggerBuilder.Create()
    .WithCronSchedule("0 0 3 * * ? *", x => x.InTimeZone(TimeZoneConverter.TZConvert.GetTimeZoneInfo("US/Eastern")))
    .Build();
```

Cron can be quite powerful tool: http://www.cronmaker.com/

*edit* fixed the hour to be 3 instead of 7 as time zone will handle it, but beware - you are again close to daylights savings...

**Reply by drusellers** [reply]
Agreed, I just misunderstood the goal of the other schedule type. Switched to cron, not a big deal there. TY!

Yeah, its ok if it misfires one to two times per year.
